core-codemods/src/main/java/io/codemodder/codemods/SensitiveDataLoggingCodemod.java

```diff
@@ -1,53 +1,205 @@
 package io.codemodder.codemods;
 
-import static io.codemodder.CodemodResources.getClassResourceAsString;
-
+import com.contrastsecurity.sarif.PhysicalLocation;
 import com.contrastsecurity.sarif.Result;
-import com.github.difflib.patch.AbstractDelta;
-import com.github.difflib.patch.DeleteDelta;
-import com.github.difflib.patch.Patch;
+import com.contrastsecurity.sarif.Run;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.stmt.Statement;
+import com.theokanning.openai.completion.chat.*;
 import io.codemodder.*;
+import io.codemodder.javaparser.JavaParserChanger;
 import io.codemodder.plugins.llm.OpenAIService;
-import io.codemodder.plugins.llm.SarifToLLMForBinaryVerificationAndFixingCodemod;
 import io.codemodder.providers.sarif.semgrep.SemgrepScan;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
 import javax.inject.Inject;
 
 /** A codemod that removes any sensitive data being logged. */
 @Codemod(
     id = "pixee:java/sensitive-data-logging",
     importance = Importance.HIGH,
     reviewGuidance = ReviewGuidance.MERGE_AFTER_REVIEW)
-public final class SensitiveDataLoggingCodemod
-    extends SarifToLLMForBinaryVerificationAndFixingCodemod {
+public final class SensitiveDataLoggingCodemod extends JavaParserChanger {
+
+  private final RuleSarif sarif;
+  private final OpenAIService service;
 
   @Inject
   public SensitiveDataLoggingCodemod(
       @SemgrepScan(ruleId = "sensitive-data-logging") final RuleSarif sarif,
-      final OpenAIService openAI) {
-    super(sarif, openAI);
+      final OpenAIService openAIService) {
+    this.sarif = sarif;
+    this.service = Objects.requireNonNull(openAIService);
   }
 
   @Override
-  protected String getThreatPrompt(
-      final CodemodInvocationContext context, final List<Result> results) {
-    return getClassResourceAsString(getClass(), "threat_prompt.txt");
+  public CodemodFileScanningResult visit(
+      final CodemodInvocationContext context, final CompilationUnit cu) {
+
+    List<Result> results = sarif.getResultsByLocationPath(context.path());
+
+    String rawFile;
+    try {
+      rawFile = Files.readString(context.path());
+    } catch (IOException e) {
+      throw new UncheckedIOException("Couldn't read file", e);
+    }
+
+    List<String> rawLines = rawFile.lines().toList();
```

Comment:
Avoid the intermediate string to save memory.

```suggestion
    final List<String> rawLines;
    try (final Stream<String> lines = Files.lines(context.path())) {
      rawLines = lines.toList();
    } catch (IOException e) {
      throw new UncheckedIOException("Couldn't read file", e);
    }
```

Also, consider combining the code that adds the line numbers with this code that streams the lines from disk, to avoid holding on to a copy of `rawLines` without line numbers that is never used.

Reply:
> Also, consider combining the code that adds the line numbers with this code that streams the lines from disk, to avoid holding on to a copy of rawLines without line numbers that is never used.

Disregard. I see now that the intention is to read the lines from disk once then take snippets from it.